packages/provider/src/transports/base-wallet-transport.ts

```diff
@@ -420,26 +417,37 @@ export abstract class BaseWalletTransport implements WalletTransport {
           }
         }
       } else {
-        // user is already connected, notify session details.
-        // TODO: in future, keep list if 'connected' dapps / sessions in the session
-        // controller, and only sync with allowed apps
-        this.notifyOpen({
-          sessionId: this._sessionId,
-          chainId: `${chainId}`,
-          session: await this.walletRequestHandler.walletSession()
-        })
+        // if user is already connected and origin is in connected dapps, notify session details.
+        if (
+          isDappConnected(this.appOrigin) ||
+          this.appOrigin?.includes('chrome-extension') ||
+          !this.walletRequestHandler.prompter
+        ) {
+          this.notifyOpen({
+            sessionId: this._sessionId,
+            chainId: `${chainId}`,
+            session: await this.walletRequestHandler.walletSession()
+          })
+        } else {
+          this.notifyOpen({
+            sessionId: this._sessionId,
+            error: ERROR_MESSAGES.DAPP_NOT_CONNECTED
+          })
+          logger.error(`dapp origin ${this.appOrigin} not found in connected dapps`)
+          return false
+        }
```

Comment:
I'm maybe not the best person to review this, but is there some explanation why it's these three conditions that call `this.notifyOpen`? Also, is the error message here correct for the conditions needed to return an error?

Reply:
Yeah I understand it's not clear as it is, need to add more comments here.

So previously we weren't checking if a dapp is connected at all. This change checks if a dapp connected with `isDappConnected`, which is the first and the main condition. Second one is to allow if this request came from extension, and last if there is no wallet request handler, in which case a dapp would not be added to connected dapps at all so we can allow connecting directly.

And about the error, in extension 'not being connected' is handled earlier, so main case this error is needed is for dapps that integrate with sequence.js directly. If user disconnects a dapp in settings, this error allows us to clear the session from that app next time they try to open the wallet.